### ISSUE TITLE
Include() gets only one document when used with Take()

### ISSUE THREAD

**Issue by luboshl** [JasperFx/marten#2433]
In the example below I need to load Product Groups related to Products. Because there are about 50k of records, I need to use paging. When I use `ToPagedListAsync()` or only `Take()` and `ToListAsync()` (`Skip` has no effect for the issue), the dictionary for related documents contains only one item even if there are 3 unique `ProductGroupId`s in my example.

```csharp
var productGroups = new Dictionary<Guid, ProductGroupDocument>();
await using var session = DocumentStore.QuerySession();

var productDocuments = await session.Query<ProductDocument>()
    .Include(x => x.ProductGroupId, productGroups)
    .Take(10)
    .ToListAsync(ct);

// 1
var productGroupsInDictionaryCount = productGroups.Count;
// 3
var uniqueProductGroupsCount = productDocuments.Select(x => x.ProductGroupId).Distinct().Count();
```

During experimenting, when I increased the number of items as `Take(1000)`, the dictionary (`productGroupsInDictionaryCount`) contained 3 items even if there was 13 unique values (`uniqueProductGroupsCount`).

**Comment by luboshl** [comment]
This issue seems to be fixed in 6.0.0-alpha.6 version.

**Comment by luboshl** [comment]
I reopen this issue, I have the same problem again, also in version 6.0.0-alpha.6. I've created PR with failing tests.
